src/autorun.ts

```diff
@@ -1,20 +1,33 @@
 import { EMPTY, Observable, of, Subject, Subscription, throwError } from 'rxjs';
 import { distinctUntilChanged, startWith, switchMap, takeWhile } from 'rxjs/operators';
 
+
+enum TrackType {
+    Weak = 0,
+    Normal = 1
+}
+
 interface TrackEntry<V> {
     hasValue: boolean;
     value?: V;
     subscription?: Subscription;
     track: boolean;
+    used: boolean;
     completed: boolean;
+    trackType: TrackType;
 }
 
 const ERROR_STUB = Object.create(null);
 
-type $FnT<T> = (o: Observable<T>) => T;
 type $Fn = <T>(o: Observable<T>) => T;
 type Cb<T> = (...args: any[]) => T;
 
+type Trackers = {
+    weak: $Fn;
+    normal: $Fn;
```

Comment:
I imagined that `$` and `$.weak` would be enough.
But if you think it's good to have — that's cool! Up to you

Reply:
My reasoning was, whatever the default is, the user can be explicit so doesn't have to rely on the default...